File: apps/ios/PrismEra/Models/Analytics.swift
import Foundation

struct OverviewStats {
    var totalViews: Int
    var viewsTrend: Double
    var engagement: Int
    var engagementTrend: Double
    var shares: Int
    var sharesTrend: Double

    static func mock() -> OverviewStats {
        OverviewStats(
            totalViews: 12580,
            viewsTrend: 12.5,
            engagement: 3420,
            engagementTrend: 8.3,
            shares: 156,
            sharesTrend: -2.1
        )
    }
}

struct ContentPerformance: Identifiable {
    let id = UUID()
    let date: Date
    let views: Int
    let engagement: Int

    static func mockData() -> [ContentPerformance] {
        let calendar = Calendar.current
        let today = Date()

        return (0..<7).reversed().map { offset in
            let date = calendar.date(byAdding: .day, value: -offset, to: today)!
            return ContentPerformance(
                date: date,
                views: Int.random(in: 500...3000),
                engagement: Int.random(in: 50...500)
            )
        }
    }
}

struct PlatformDistribution: Identifiable {
    let id = UUID()
    let platform: String
    let percentage: Double

    static func mockData() -> [PlatformDistribution] {
        [
            PlatformDistribution(platform: "微信", percentage: 35),
            PlatformDistribution(platform: "小红书", percentage: 28),
            PlatformDistribution(platform: "微博", percentage: 20),
            PlatformDistribution(platform: "知乎", percentage: 12),
            PlatformDistribution(platform: "其他", percentage: 5)
        ]
    }
}

struct TopContent: Identifiable {
    let id = UUID()
    let title: String
    let views: Int
    let growth: Double

    static func mockData() -> [TopContent] {
        [
            TopContent(title: "AI 内容创作完全指南", views: 3520, growth: 25.3),
            TopContent(title: "2026 营销趋势预测", views: 2890, growth: 18.7),
            TopContent(title: "新手入门教程", views: 2150, growth: 12.1),
            TopContent(title: "效率提升秘籍", views: 1890, growth: 8.5)
        ]
    }
}

File: apps/android/PrismEra/app/src/main/java/com/prismera/app/data/model/Analytics.kt
package com.prismera.app.data.model

import java.util.*

data class OverviewStats(
    val totalViews: Int,
    val viewsTrend: Double,
    val engagement: Int,
    val engagementTrend: Double,
    val shares: Int,
    val sharesTrend: Double
)

data class ContentPerformance(
    val date: Long,
    val views: Int,
    val engagement: Int
) {
    companion object {
        fun mockData(): List<ContentPerformance> {
            val calendar = Calendar.getInstance()
            val today = System.currentTimeMillis()
            return (0..6).reversed().map { offset ->
                val date = calendar.apply { add(Calendar.DAY_OF_MONTH, -offset) }.timeInMillis
                ContentPerformance(
                    date = date,
                    views = (500..3000).random(),
                    engagement = (50..500).random()
                )
            }
        }
    }
}

data class PlatformDistribution(
    val platform: String,
    val percentage: Double
) {
    companion object {
        val sampleData = listOf(
            PlatformDistribution("微信", 35.0),
            PlatformDistribution("小红书", 28.0),
            PlatformDistribution("微博", 20.0),
            PlatformDistribution("知乎", 12.0),
            PlatformDistribution("其他", 5.0)
        )
    }
}

data class TopContent(
    val title: String,
    val views: Int,
    val growth: Double
) {
    companion object {
        val sampleData = listOf(
            TopContent("AI 内容创作完全指南", 3520, 25.3),
            TopContent("2026 营销趋势预测", 2890, 18.7),
            TopContent("新手入门教程", 2150, 12.1),
            TopContent("效率提升秘籍", 1890, 8.5)
        )
    }
}
